src/main/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraAllActionScenario.kt

```diff
@@ -0,0 +1,130 @@
+package com.atlassian.performance.tools.jiraactions.api.scenario
+
+import com.atlassian.performance.tools.jiraactions.api.SeededRandom
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import com.atlassian.performance.tools.jiraactions.api.action.*
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import com.atlassian.performance.tools.jiraactions.api.memories.IssueKeyMemory
+import com.atlassian.performance.tools.jiraactions.api.memories.adaptive.AdaptiveIssueKeyMemory
+import com.atlassian.performance.tools.jiraactions.api.memories.adaptive.AdaptiveIssueMemory
+import com.atlassian.performance.tools.jiraactions.api.memories.adaptive.AdaptiveJqlMemory
+import com.atlassian.performance.tools.jiraactions.api.memories.adaptive.AdaptiveProjectMemory
+
+/**
+ * Provides Jira `Scenario`.
+ * This class returns all available actions at current time.
+ * It is highly possible and expected that over time these actions
+ * will be different. Thus <strong>don't</strong> assume
+ * actions provided by this class will never change.
+ * @since 3.6.0
+ */
+class JiraAllActionScenario constructor() : Scenario {
```

Comment:
Am I correct in my understanding that the only reason why we're creating this class is because we don't have access to `issueKeyMemory` outside of this module?
 If so, can't we just create a custom scenario similar to the way we're creating [JiraSoftwareScenario](https://github.com/atlassian/jira-software-actions/blob/b5e9bda7845bdeea4dc52568de52714b8ca1cb3c/src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/api/JiraSoftwareScenario.kt#L31) ?

Reply:
We have acceess to issueKeyMemory. But the scenario itself has own memory, and the constructor is used to provide externally initialized (then the built in one is not used),